src/components/split-panel/index.js

```diff
@@ -0,0 +1,98 @@
+import { useState, useRef } from 'react';
+
+/**
+ * A design agnostic user resizable split panel controller.
+ * @param {function} children A `function(props)` for rendering the panels and handle (see implementation for props)
+ * @param {?number} [splitDefault=0.65] A number [0...1] as the default % split position
+ * @param {?number} [splitMin=0] A number [0...1] as the minimum % split position
+ * @param {?number} [splitMax=1] A number [0...1] as the maximum % split position
+ * @param {?number} [keyboardStep=0.025] A number [0...1] as the % step to move split when using keyboard
+ * @param {?string} [orientation='vertical'] Only 'vertical' currently supported.
+ * @return {object} The rendered children
+ **/
+export const SplitPanel = ({
+  splitDefault = 0.65,
+  splitMin = 0,
+  splitMax = 1,
+  keyboardStep = 0.025,
+  orientation = 'vertical',
+  children,
+}) => {
+  const containerRef = useRef();
+  const handleRef = useRef();
+
+  const getRects = () => ({
+    container: containerRef.current?.getBoundingClientRect(),
+    handle: handleRef.current?.getBoundingClientRect(),
+  });
+
+  const clampSplit = (value) => {
```

Comment:
I know it's hard to find a good variable name for this, but `clampSplit` sounds a bit confusing - perhaps calling it 'calculateSplit` or `calculateXX` ( whatever it is trying to calculate) would be more intuitive?

Reply:
Fair enough if `clamp` seems to clash with `split` I guess, I've refactored this to just `clamp` then.